src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Property.cs

```diff
@@ -39,5 +45,11 @@ public ExprProperty(CType type, Expr pOptionalObjectThrough, Expr pOptionalArgum
         public MethWithType MethWithTypeSet { get; }
 
         public override SymWithType GetSymWithType() => PropWithTypeSlot;
+
+        internal static bool HasIsExternalInitModifier(MethWithType mwtSet)
+        {
+            var types = (mwtSet.Meth()?.AssociatedMemberInfo as MethodInfo)?.ReturnParameter.GetRequiredCustomModifiers();
+            return (types != null) && types.Any(type => type.FullName == "System.Runtime.CompilerServices.IsExternalInit");
```

Comment:
Know the compiler matches by name here but do we check to make sure it is a `class` in the compiler? Or do we allow the modreq to be any kind of type?

Reply:
The compiler doesn't appear to check for `class`.

The compiler does have some additional checks though. Updated here.